java/feathub-udf/flink-udf/src/main/java/com/alibaba/feathub/flink/udf/PeriodicEmitLastValueOperator.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2022 The FeatHub Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.feathub.flink.udf;
+
+import org.apache.flink.api.common.operators.ProcessingTimeService;
+import org.apache.flink.api.common.state.ListState;
+import org.apache.flink.api.common.state.ListStateDescriptor;
+import org.apache.flink.runtime.state.StateInitializationContext;
+import org.apache.flink.streaming.api.operators.AbstractStreamOperator;
+import org.apache.flink.streaming.api.operators.BoundedOneInput;
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.types.Row;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.Iterator;
+
+/** An operator that periodically emits the last value it has received from upstream operator. */
+public class PeriodicEmitLastValueOperator extends AbstractStreamOperator<Row>
+        implements OneInputStreamOperator<Row, Row>, Serializable, BoundedOneInput {
+    private final long emitIntervalMs;
+
+    private transient ListState<Row> state;
+
+    public PeriodicEmitLastValueOperator(Integer emitIntervalMs) {
+        this.emitIntervalMs = emitIntervalMs.longValue();
+    }
+
+    @Override
+    public void initializeState(StateInitializationContext context) throws Exception {
+        super.initializeState(context);
+        state =
+                context.getOperatorStateStore()
+                        .getListState(new ListStateDescriptor<>("lastRowState", Row.class));
+    }
+
+    @Override
+    public void processElement(StreamRecord<Row> element) throws Exception {
+        if (!state.get().iterator().hasNext()) {
```

Comment:
This introduces one state access for each processElement call. Can we avoid the state access by introducing a flag in memory to keep track of whether the state is empty? 

And we can initialize the flag during state initialization.